micrometer-core/src/main/java/io/micrometer/core/instrument/binder/okhttp3/OkHttpContext.java

```diff
@@ -24,14 +24,17 @@
 
 import java.util.function.BiFunction;
 import java.util.function.Function;
+import java.util.function.Supplier;
 
 /**
  * A {@link SenderContext} for OkHttp3.
  *
  * @author Marcin Grzejszczak
  * @since 1.10.0
  */
-public class OkHttpContext extends RequestReplySenderContext<Request.Builder, Response> {
+@SuppressWarnings("jol")
+public class OkHttpContext extends RequestReplySenderContext<Request.Builder, Response>
+        implements Supplier<OkHttpContext> {
```

Comment:
Demonstrates one way to deal with this breaking change for users. Then they can pass the, in this case, OkHttpContext instead of using a lambda.